.github/workflows/ci.yml

```diff
@@ -181,6 +181,18 @@ jobs:
       - name: "test --workspace --all-features"
         run: cargo check --workspace --all-features
 
+  minimal-versions:
+    name: minimal-versions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env.nightly }}
+          override: true
+      - name: "check --all-features -Z minimal-versions"
+        run: cargo check --all-features -Z minimal-versions
```

Comment:
If I remember correctly, `cargo check --Z mimimal-version` doesn't really do anything. It needs to be separated into `cargo update --Z mimimal-version` and `cargo check`.
Also, dev-dependencies may raise version requirements. Ideally, remove them before run `cargo update --Z mimimal-version`. (Also, note that `Cargo.lock` is actually shared within the workspace. However as far as I know, there is no workaround for this yet.)

So, I prefer to use commands like:
```sh
# Remove dev-dependencies from Cargo.toml to prevent the next `cargo update`
# from determining minimal versions based on dev-dependencies.
cargo hack --remove-dev-deps --workspace
# Update Cargo.lock to minimal version dependencies.
cargo update -Z minimal-versions
cargo check --all-features
```

(See also [`pin-project`'s `check-minimal-version` script](https://github.com/taiki-e/pin-project/blob/b3210d10822400f694e916f570d1f9464494747d/scripts/check-minimal-versions.sh))

Reply:
I'll try that.